airflow/api/common/experimental/delete_dag.py

```diff
@@ -42,7 +42,10 @@ def delete_dag(dag_id: str, keep_records_in_log: bool = True, session=None) -> i
     """
     log.info("Deleting DAG: %s", dag_id)
     running_tis = (
-        session.query(models.TaskInstance.state).filter(models.TaskInstance.state.in_(State.unfinished)).all()
+        session.query(models.TaskInstance.state)
+        .filter(models.TaskInstance.dag_id == dag_id)
+        .filter(models.TaskInstance.state == State.RUNNING)
+        .first()
```

Comment:
Ash mentioned `exists()` would be more (most, I think) performant https://github.com/apache/airflow/pull/17630#discussion_r689343499

But if you don’t want to deal with the MSSQL caveat, `count()` is probably the next best.

Reply:
first (`LIMIT 1`) would be less work on the DB than ` COUNT()` I think

Reply:
I added count with limit. With only `limit`, other tests failed.